add option for removing testcases without the test_key property

devilofhell opened the pull request:
This commit removes all testcases from the report, that do not contain the "test_key" property - if the option in playwright is set.

when running my testsuite of 100+ tests, I mark most of the tests with a test_key property. However, if a test does not contain that property, the import to xray-cloud fails. Since I don't want to go through the xml report myself to remove all the tests without a test_key (and I will always have some tests without a test_key like global-setup tests), I want to throw out all testcases I dont need - thus this PR. 
I also added an optional Regex, that the test_key property needs to match, and filters out all testcases that do not match aswell.

example 

```typescript
const xrayOptions = {
    embedAnnotationsAsProperties: true,
    textContentAnnotations: ["test_description"],
    outputFile: "./report-export/xray-report.xml",

    removeTestCasesWithoutTestKey: true,  // default is false
    requiredTestKeyRegex: /^ABC-\d+$/,  // "ABC-" as a string also works. has no effect if previous option is set to false
};

export default defineConfig({
    reporter: [
        ["@xray-app/playwright-junit-reporter", xrayOptions],
    ],
}
```

I really do need this feature and I planned on using my own fork, but maybe you are also interested in it and I dont need to maintain my own fork.

Please bear with me, this is the first Pull Request I am doing to a public repostory, feedback is welcome <3

bitcoder commented:
Hi @devilofhell , first of all, thanks for your contribution!
It would be preferable to start with a feature request, by opening an issue, and then do the PR after discussing and agreeing on it .
I see the value of this, but just one question: why having this `requiredTestKeyRegex` setting? What's the need we're trying to address with it? I'm just trying to understand the usecase in order to avoid having unnecessary options, if that's the case

devilofhell commented:
Thank you for your kind feedback! I'll open an issue first, next time, i need a feature.

The `requiredTestKeyRegex` option is for filtering the test report to only import project-based tests. 
Imagine you have 2 projects in your Jira ("ABC", "XYZ") and both have seperate Issue tracking ("ABC-1", "ABC-2", and "XYZ-1", "XYZ-2" etc. Further lets say your Playwright project does have tests for both projects, but you want to import 2 reports, for each project respectively Testexecution("ABC-123") and TestExecution("XYZ-456"). 

Or you could catch typos, if you write your testKey wrong into the property ("ACB-43")

-------

However, ...
thinking of such use cases, its propably better to organize your test-suite different, like having those different tests sitting in different playwright-projects. 
And if you actually mistyped your testKey, its propably better to have Xray throw an error during import, instead of deleting the test from the report.

The more i think about it, the more I think, that this option is overkill and I should remove this option (as that option is actually not necessary to me, it was just a quick-win).

bitcoder commented:
I agree on just having the `removeTestCasesWithoutTestKey` option.
Can you please update the PR to include tests also, testing the case when it's true and false and missing?
Also, updating the doc on the README.md.
Thanks!

devilofhell commented:
Hi,
I updated my PR.

Since I moved the filtering to another place (to also have the testcount inside the suite fit), I also renamed the option from `removeTestCasesWithoutTestKey` to `ignoreTestCasesWithoutTestKey`, because thats now more accurate in terms of "whats happening in the code". 
Is the documenation inside the README satisfying?

devilofhell commented:
Done, 
thank you :)
